### PR TITLE
Add more notify call for scoring

### LINKED ISSUE #2597: chore: Add more notify call for scoring
### Background
This issue captures follow up task after https://github.com/cowprotocol/services/pull/2585 is merged.

Few more notify types should be added: `InvalidExecutedAmount` and `MissingPrices` are probably worth adding.

### PR DESCRIPTION
**Pull request by Mateo-mro**
# Description
Add more notify call for scoring

# Changes
Add notify calls for the following score errors:
- `InvalidExecutedAmount`
- `MissingPrice(TokenAddress)`
- `CustomPrice(String)`

## How to test
1. e2e
2. unit test

## Related Issues
Fixes #2597

### CODE REVIEW

File: crates/driver/src/domain/competition/solution/mod.rs [hunk 1]
```diff
@@ -497,7 +499,31 @@ pub mod error {
         InvalidClearingPrices,
         #[error(transparent)]
         Math(#[from] Math),
+        #[error("failed to calculate custom prices")]
+        CalculateCustomPrices(#[source] Trade),
+        #[error("missing native price for token {0:?}")]
+        MissingPrice(TokenAddress),
+        #[error("failed to calculate custom price {0:?}")]
```
Comment (MartinquaXD) [inline crates/driver/src/domain/competition/solution/mod.rs:506]: What makes this error different from `CalculateCustomPrices`? From the perspective of the solver they seem pretty indistinguishable and not self explanatory.

**Review by MartinquaXD — COMMENTED** [review]
I think those new error types also have to be added to the parsing logic on the solver side.

File: crates/driver/src/domain/competition/solution/mod.rs (hunk 1 above)
Comment (Mateo-mro) [reply crates/driver/src/domain/competition/solution/mod.rs:506]: Fair enough, that error could be squashed into its upper level error `error::Scoring` without a problem.

File: crates/driver/src/domain/competition/solution/mod.rs [hunk 2]
```diff
@@ -497,7 +499,29 @@ pub mod error {
...
+        fn from(value: scoring::Error) -> Self {
+            match value {
+                scoring::Error::MissingPrice(e) => Self::MissingPrice(e),
+                scoring::Error::Math(e) => Self::Math(e),
+                scoring::Error::Scoring(e) => e,
+            }
+        }
+    }
+
+    #[derive(Debug, thiserror::Error)]
```
Comment (MartinquaXD) [inline crates/driver/src/domain/competition/solution/mod.rs:518]: Why move this here?
Looks like the error is mainly used in a different file and we only need it to know about it here to convert from it.

File: crates/driver/src/infra/notify/notification.rs [hunk 3]
```diff
@@ -51,6 +51,10 @@ pub enum Kind {
 pub enum ScoreKind {
     /// No clearing prices are present for all trades.
     InvalidClearingPrices,
+    /// Trade: invalid executed amount
+    InvalidExecutedAmount,
+    /// missing native price for token {0:?}
```
Comment (MartinquaXD) [inline crates/driver/src/infra/notify/notification.rs:56]: These doc comments could be more explicit. Also why use the debug notation here?

File: crates/driver/src/infra/solver/dto/notification.rs [hunk 4]
```diff
@@ -108,6 +118,10 @@ pub enum Kind {
         succeeded_once: bool,
     },
     InvalidClearingPrices,
+    MissingPrice {
```
Comment (MartinquaXD) [inline crates/driver/src/infra/solver/dto/notification.rs:121]:
```suggestion
    #[serde(rename_all = "camelCase")]
    MissingPrice {
```

**Review by MartinquaXD — COMMENTED** [review]
I think we still need a change in the `solvers-dto` crate. Otherwise the solver will not know how to deserialize the new error. This will then also require an update in the gnosis `solvers` repo.

File: crates/driver/src/infra/solver/dto/notification.rs (hunk 4 above)
Comment (Mateo-mro) [reply crates/driver/src/infra/solver/dto/notification.rs:121]: This is redundant. The enum `Kind` already derives it.

Comment (MartinquaXD) [reply crates/driver/src/infra/solver/dto/notification.rs:121]: The `derive` on the enum kind only causes the enum variant **names** to be camelcased but not the **fields** of the structs inside the variant.

Comment (Mateo-mro) [reply crates/driver/src/infra/solver/dto/notification.rs:121]: ah, I misunderstood, in that case yes.

**Review by squadgazzz — APPROVED** [review]

**Review by MartinquaXD — APPROVED** [review]
Looks okay to me.
@sunce86 could you also give it a look. I think since you worked quite a bit on the notification system you are probably more likely to spot errors than anyone of use.

File: crates/driver/src/domain/competition/solution/scoring.rs [hunk 5]
```diff
@@ -419,6 +419,6 @@ pub enum Error {
     MissingPrice(eth::TokenAddress),
     #[error(transparent)]
     Math(#[from] Math),
-    #[error("failed to calculate custom price {0:?}")]
-    CustomPrice(String),
+    #[error("scoring: failed to calculate custom price {0:?}")]
```
Comment (sunce86) [inline crates/driver/src/domain/competition/solution/scoring.rs:422]: i believe this comment needs to be updated

**Review by sunce86 — APPROVED** [review]

**Pull request merged**